src/definitions/session.rs

```diff
@@ -199,7 +199,10 @@ pub fn derive_session_key(
     session_transcript: &SessionTranscriptBytes,
     reader: bool,
 ) -> Result<GenericArray<u8, U32>> {
-    let salt = Sha256::digest(serde_cbor::to_vec(session_transcript)?);
+    let salt = Sha256::digest(
+        crate::cbor::to_vec(session_transcript)
+            .map_err(|e| anyhow::anyhow!("failed to serialize session transcript: {e}"))?,
```

Comment:
import `anyhow::Context` and use:

```suggestion
        crate::cbor::to_vec(session_transcript)
            .context("failed to serialize session transcript")?,
```